<a href="https://colab.research.google.com/github/vthaanesh22/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/vthaanesh22/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [7]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


I chose the Refresh / Content Opportunity Scoring lane because I want to find which pages should be reviewed first for improvement using the available data.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The decision I want to improve is which pages should be reviewed first. A content or search team member would use the results to decide whether a page should be refreshed, improved, protected, pruned, or monitored. A wrong recommendation could waste limited review time on a page that does not need attention, while missing an important page could cause a useful improvement opportunity to be missed.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

url = "https://raw.githubusercontent.com/vthaanesh22/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of pages:", len(df))
print("Number of clients:", df["client_id"].nunique())
print("Pages with downward trend:", (df["trend_direction"] == "down").sum())

Number of pages: 30000
Number of clients: 32
Pages with downward trend: 16262


The starter dataset contains 30,000 pages from 32 clients. I found that 16,262 pages have a downward trend in the available data. This makes the refresh and content opportunity lane worth investigating because there is a substantial number of pages that may deserve further review. However, the downward trend is based on the current data window, so it should be treated as an initial signal rather than proof of future decline.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


This project can provide observed, measured, and directional evidence from the available data. The results can be used as decision-support to help prioritize pages for human review. I cannot claim that refreshing a page will cause it to recover, because this analysis does not provide causal proof. I also cannot claim to predict or explain Google's ranking algorithm.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.